libraries/apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/network/http/HttpNetworkTransport.kt

```diff
@@ -100,14 +103,32 @@ private constructor(
       operation: Operation<D>,
       throwable: Throwable,
   ): ApolloResponse<D> {
-    val apolloException = if (throwable is ApolloException) {
-      throwable
-    } else {
-      // This happens for null pointer exceptions on missing fields
-      ApolloParseException(
-          message = "Failed to parse GraphQL http network response",
-          cause = throwable
-      )
+    val apolloException = when(throwable) {
+      is ApolloException -> {
+        // This happens for malformed JSON
+        throwable
+      }
+      is EOFException -> {
+        // This happens for premature end of file
+        ApolloParseException(
+            message = "EOFException while parsing the HTTP network response",
+            cause = throwable
+        )
```

Comment:
This is an interesting one. It could be argued that EOFException is a network exception if the stream is closed too early

* If there is a `Content-Length` this doesn't happen because OkHttp has a specific `java.net.ProtocolException` for this
* I'm not 100% sure about chunked encoding
* Also not 100% clear about the other engines (Ktor, etc...)